llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -2752,10 +2762,10 @@ void VPWidenLoadRecipe::execute(VPTransformState &State) {
     NewLI = Builder.CreateAlignedLoad(DataTy, Addr, Alignment, "wide.load");
   }
   // Add metadata to the load, but setVectorValue to the reverse shuffle.
-  State.addMetadata(NewLI, LI);
-  if (Reverse)
-    NewLI = Builder.CreateVectorReverse(NewLI, "reverse");
-  State.set(this, NewLI);
+  State.addNewMetadata(NewLI, LI);
+  applyMetadata(*NewLI);
+  State.set(this,
+            Reverse ? Builder.CreateVectorReverse(NewLI, "reverse") : NewLI);
```

Comment:
This change is to avoid casting the vector reverse of NewLI to Instruction:
```
  if (Reverse)
    NewLI = cast<Instruction>(Builder.CreateVectorReverse(NewLI, "reverse"));
  State.set(this, NewLI);
```
?

Reply:
Yes, otherwise we need to cast, could also change if desired

Reply:
restored thanks